### PR TITLE
Careers updates

### PR DESCRIPTION
**Pull request by cwackerfuss**
Three updates:
1. Moves the Careers link to the site header and says "We're Hiring!"
2. Satisfies https://github.com/blockstack/blockstack.org/issues/505
3. Satisfies https://github.com/blockstack/blockstack.org/issues/509 by adding the Lever items to the Careers page. To give them more priority on the page, I introduced a responsive sidebar.

Walkthrough:
careers-work

Demonstrating responsiveness:
careers-work-responsiveness

### CODE REVIEW

**Comment by guylepage3** [comment]
Nice work here @cwackerfuss 👍

File: app/styles/main.css [hunk 1]
```diff
@@ -2843,6 +2847,12 @@ pre {
     padding-bottom: 50px;
   }
 
+  /* Jobs Sidebar */
```
[the remark this answers is not in the document]
Comment (cwackerfuss) [reply app/styles/main.css]: @guylepage3 do you mean move them in the css file, or apply styles to other sidebars too?

[the remark this answers is not in the document]
Comment (guylepage3) [reply app/styles/main.css]: @cwackerfuss sorry. I mean to move them to this location.. https://github.com/blockstack/blockstack.org/blob/ad74b9145153434008faa7ef117e6709d8350bff/app/styles/main.css#L2108

[the remark this answers is not in the document]
Comment (guylepage3) [reply app/styles/main.css]: STET. Nevermind. These styles are located in a media query

**Pull request merged**